app/src/main/java/sw19/moring03/paint/Views/DrawingView.java

```diff
@@ -54,32 +57,38 @@ public boolean onTouchEvent(MotionEvent event) {
                 invalidate();
                 break;
             case MotionEvent.ACTION_MOVE:
-                addPoint(xCoord, yCoord);
+                if (((MainActivity) getContext()).getChosenTool() != Tool.FILL) {
```

Comment:
please override the addPoint method in the FillTool itself and set the logic there that the point should not get added when already a point is in the point list of the tool

Reply:
Fixed with commit 7225edaa72e